workspace_host/interface.js

```diff
@@ -918,19 +921,39 @@ async function sendGradedFilesArchive(workspace_id, res) {
   const zipName = `${workspace.remote_name}-${timestamp}.zip`;
   const workspaceDir = path.join(config.workspaceHostHomeDirRoot, workspace.remote_name, 'current');
 
+  const gradedFiles = (
+    await fg(workspaceSettings.workspace_graded_files, {
+      cwd: workspaceDir,
+      stats: true,
+    })
+  ).filter((file) => contains(workspaceDir, path.join(workspaceDir, file.path)));
+
+  if (gradedFiles.length > config.workspaceMaxGradedFilesCount) {
+    return res.status(500).send({
+      message: `Cannot submit more than ${config.workspaceMaxGradedFilesCount} files from the workspace.`,
+    });
+  }
+  if (_.sumBy(gradedFiles, (file) => file.stats.size) > config.workspaceMaxGradedFilesSize) {
+    return res.status(500).send({
+      message: `Workspace files exceed limit of ${filesize(config.workspaceMaxGradedFilesSize, {
+        base: 2,
+      })}.`,
+    });
+  }
+
   // Stream the archive back to the client as it's generated.
   res.attachment(zipName).status(200);
   const archive = archiver('zip');
   archive.pipe(res);
 
-  for (const file of workspaceSettings.workspace_graded_files) {
+  for (const file of gradedFiles) {
     try {
-      const filePath = path.join(workspaceDir, file);
+      const filePath = path.join(workspaceDir, file.path);
       await fsPromises.lstat(filePath);
```

Comment:
We don't seem to be using the result of this call?

Reply:
Removed in 797286c.